packages/components/src/components/Signin/Signin.tsx

```diff
@@ -0,0 +1,26 @@
+import { ethers } from 'ethers';
+import React from 'react';
+import { useWallet, Provider } from '@web3-ui/hooks';
+import { Button } from '@chakra-ui/react';
+export interface SigninProps {
+  data: string;
```

Comment:
Seems like `data` can hold different types, not just a string, could we use a generic to allow folks to pass what is the expected data?

Reply:
so data is the message the user signs in their wallet but it should be named more clearly.